Sources/Jobs/Scheduler/DateExtensions/DateExtensionRetrival.swift

```diff
@@ -0,0 +1,166 @@
+import Foundation
+
+extension Date {
+    func second() -> Int? {
+        return Calendar.current.dateComponents([.second], from: self).second
+    }
+
+    func minute() -> Int? {
+        return Calendar.current.dateComponents([.minute], from: self).minute
+    }
+
+    func hour() -> Int? {
+        return Calendar.current.dateComponents([.hour], from: self).hour
+    }
+
+    func dayOfWeek() -> Int? {
+        return Calendar.current.dateComponents([.weekday], from: self).weekday
+    }
+
+    func dayOfMonth() -> Int? {
+        return Calendar.current.dateComponents([.day], from: self).day
+    }
+
+    func weekOfMonth() -> Int? {
+        return Calendar.current.dateComponents([.weekOfMonth], from: self).weekOfMonth
+    }
+
+    func weekOfYear() -> Int? {
+        return Calendar.current.dateComponents([.weekOfYear], from: self).weekOfYear
+    }
+
+    func month() -> Int? {
+        return Calendar.current.dateComponents([.month], from: self).month
+    }
+
+    func quarter() -> Int? {
+        // Bug: doesn't work in ios 12 and macOS Mojave 10.14
+        //return Calendar.current.dateComponents([.quarter], from: self).quarter
```

Comment:
If this doesn't work on Mojave either we should probably leave the workaround. We can remove these comments for now